Spigot-API-Patches/0291-Add-hasMoved-to-PlayerMoveEvent.patch

```diff
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: HexedHero <6012891+HexedHero@users.noreply.github.com>
+Date: Thu, 29 Apr 2021 10:31:44 +0100
+Subject: [PATCH] Add hasMoved to PlayerMoveEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java b/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java
+index 1a58734d919fae247eeb85dd785fd59990856505..a25556f52461b4e77b135c9f9fd30e3b6afdb08d 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerMoveEvent.java
+@@ -93,6 +93,18 @@ public class PlayerMoveEvent extends PlayerEvent implements Cancellable {
+         this.to = to;
+     }
+ 
++    // Paper start
++    /**
++     * Check if the player has actually moved somewhere and not just
++     * changed looking direction which also fires the event
++     *
++     * @return whether the player has moved or not
++     */
++    public boolean hasMoved() {
```

Comment:
I think this name is slightly misleading.. Maybe something like `hasMovedBody`? I'm not sure tbh

Reply:
I feel that is just as misleading because your body moves when you turn.

Reply:
Maybe `hasMovedToAnotherBlock` would be more appropriate ? Or something shorter if you find one.

Reply:
That's also kinda problematic because you can still move within the same block.

Reply:
Oh yes i thougth the getters where `getBlockX/Y/Z` ones

Reply:
`#hasChangedPosition` and `#hasChangedOrientation`.

Reply:
Maybe another method `#hasChangedBlock` would be relevent to check if the `to` block is different of the `from` block